docs/DeveloperGuide.md

```diff
@@ -1308,3 +1312,52 @@ testers are expected to do more *exploratory* testing.
+
+    Lastly, we had to integrate all these models with the existing code to save persons to a data file in AB3. This required the creation of many classes and major refactoring of existing classes to support multiple models.
+
+2. **Redesigned GUI**
+
+    Compared to AB3, CoLAB has more than double the number of UI components. Each of these components had to be painstakingly designed and styled.
+
+    In addition, in line with our focus on user experience, we had to design a new `UiCommand` class to allow each command to display their own combination of UI components.
+
+    We also had to implement a combination of listeners to ensure CoLAB's UI is also navigable using a mouse. This involved a complex combination of event listeners to ensure that the behaviour of each button is consistent with the corresponding command, and the correct buttons or list cells are highlighted in the side panel.
+
+    Each of the Ui components are also responsive and work on a large range of screen sizes. To handle edge cases, we had to find a way to allow horizontal scrolling of each component individually without affecting the ability to scroll the entire window vertically.
+
+3. **Automated GUI Testing**
+
+    We implemented automated GUI testing using the TestFX library. Although we were able to reference AB4's codebase, we still had to spend many hours integrating it with our project because,
+        - Libraries in AB4 were outdated and had to be upgraded to the newest versions.
+        - As AB4 did not use github actions, we had to spend time to make sure automated GUI tests ran properly with github actions.
+        - Since our GUI and Model were so different compared to AB4, we had to spend countless hours adapting AB4's code for our project. We also had to write our own GUI tests, which took up a lot more effort than expected.
+
+4. **CRUD operations**
+
+    We had to implement create, read, update, and delete (CRUD) operations for each of the new models we created. This was not as simple due to the complexity and size of our models. For example, we had to ensure that the operations were executed on the correct task as the index shown to the user may not be the same as the index in the list (as the list displayed to the user is sorted). We also had to ensure that the many lists were all updated correctly and kept in sync with each other.
+
+5. **Undo/Redo**
+
+    We implemented the `Undo/Redo` feature in CoLAB. Unlike other similar projects where the `Undo/Redo` command simply reverts the data to the previous state, we wanted our Undo/Redo command to keep track of the state of the UI when a command is executed. This meant that we could not take reference from existing projects and had to design our own implementation of the `Undo/Redo` command.
+
+    In addition, since many of our commands did not change any data, we had to design a way to save only selected commands.
```

Comment:
Perhaps change `save only selected commands` to `save data only after excuting some selected commands`?